### PR TITLE
PMM-7605 Move pmm to the github actions

### PR DESCRIPTION
**Pull request by eoyy**

### CODE REVIEW

**Review by ritbl — APPROVED** [review]
LGTM

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          - version: tip
+            may-fail: true
+
+    continue-on-error: ${{ matrix.go.may-fail }}
+
+    runs-on: ubuntu-latest
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      WORKING_DIR: ${{ github.workspace }}/src/github.com/percona/pmm
```
Comment (artemgavrilov) [inline .github/workflows/go.yml:34]: We don't use it in other CI workflows. Why do we need it here? It's used in just one place.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
+name: build
+
+on:
+  push:
+    branches:
+      - main
+      - release/*
+      # skip non-trunk PMM-XXXX/SAAS-XXXX branch builds, but still build pull requests
+      - '!PMM-[0-9]+'
+      - '!SAAS-[0-9]+'
```
Comment (artemgavrilov) [inline .github/workflows/go.yml:10]: But why do we need negative conditions here? If we will drop these lines only `main` and `release` will trigger build, no?

Comment (eoyy) [reply .github/workflows/go.yml:10]: it was migrated from travis.yml 
it ignores (probably) only branches with no description in the name. e.g. the current branch has the description.
Do you think this condition is outdated?

Comment (artemgavrilov) [reply .github/workflows/go.yml:10]: I think yes, we are using something like this in all other repos: https://github.com/percona/pmm-managed/blob/58e8a4f6b8f5a1a3b5ba4c1817e757eb388c20bc/.github/workflows/ci.yml#L3-L15

File: .github/workflows/cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          path: ~/.cache/go-build
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-${{ hashFiles('**') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-
+      - name: Clean Go modules cache
+        run: go clean -modcache
+
+      - name: Clean Go build cache
+        run: go clean -cache
```
Comment (artemgavrilov) [inline .github/workflows/cleanup.yml:76]: Please add newline

File: .github/workflows/go.yml (hunk 2 above)
Comment (eoyy) [reply .github/workflows/go.yml]: Ok, I fixed this.

File: .github/workflows/go.yml (hunk 1 above)
Comment (eoyy) [reply .github/workflows/go.yml:33]: It's needed to run unit tests correctly.

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+          echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
+          echo "$HOME/gotip/bin" >> $GITHUB_PATH
+
+      - name: Enable Go modules cache
+        uses: percona-platform/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-
```
[the remark this answers is not in the document]
Comment (eoyy) [reply .github/workflows/go.yml:67]: Done

**Review by artemgavrilov — APPROVED** [review]

**Comment by artemgavrilov** [comment]
Can you please also rename `go.yml` to `ci.yml` as we name this file in other repos?

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
```
Comment (artemgavrilov) [inline .github/workflows/go.yml:68]: I see that modules cached is not used for some reason and modules are downloaded each time.
Let's add here explicit step to resolve modules
```
      - name: Download Go modules
        run: go mod download
```

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    runs-on: ubuntu-latest
+    env:
+      SLACK_WEBHOOK: ${{ secrets.SLACK_TOKEN }}
+      SLACK_CHANNEL: "pmm-ci"
+      SLACK_USERNAME: "PR pipelines bot"
+      SLACK_ICON_EMOJI: ":chestnut:"
+      SLACK_COLOR: "#FF0000"
+      SLACK_TITLE: "Finished ${{ github.event.repository.name }} workflow"
+      SLACK_MESSAGE: "Workflow failed: ${{ github.event.inputs.repo || github.repository }}:${{ github.event.inputs.branch || github.head_ref }}"
+      SLACK_FOOTER: "Please check the Actions URL ^"
```
Comment (atymchuk) [inline .github/workflows/go.yml:127]: Can we remove `SLACK_FOOTER: Please check the Actions URL ^`? Looks redundant and just clutters the message.

**Comment by eoyy** [comment]
> Can you please also rename `go.yml` to `ci.yml` as we name this file in other repos?

Done

**Review by artemgavrilov — DISMISSED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Review by atymchuk — APPROVED** [review]

**Pull request merged**